extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -29,85 +75,74 @@ protected function setGlobalDisplayVars() {
-		$newestTs = $this->wg->Memc->get( $user_key );
-
-		if ( $this->ts > $newestTs ) {
-			WikiaLogger::instance()->info( 'ImageReviewLog', [
-				'method' => __METHOD__,
-				'message' => "I've got the newest ts ({$this->ts}), I won't refetch the images",
-			]);
-			$this->imageList = array();
-			$this->wg->memc->set( $user_key, $this->ts, 3600 /* 1h */ );
-		} else {
-			$this->imageList = $this->helper->refetchImageListByTimestamp( $this->ts );
+		if ( $this->wg->request->wasPosted() ) {
+			$this->handlePostAction();
 		}
 
-		/* SUS-541 / Mix <mix@wikia.com> / scope: the following if block */
-		if ( count( $this->imageList ) < ImageReviewHelper::LIMIT_IMAGES ) {
-			$this->logImageListCompleteness( 'warning' );
+		if ( $this->checkRedirect() ) {
+			// Return false to stop any template rendering
+			return false;
 		}
 
-		if ( count( $this->imageList ) == 0 ) {
-			$do = array( 
-				self::ACTION_QUESTIONABLE	=> ImageReviewStatuses::STATE_QUESTIONABLE,
-				self::ACTION_REJECTED		=> ImageReviewStatuses::STATE_REJECTED,
-				'default'					=> ImageReviewStatuses::STATE_UNREVIEWED
-			);
-
-			if ( isset( $do[ $this->action ] ) ) {
-				$this->imageList = $this->helper->getImageList( $this->ts, $do[ $this->action ], $this->order );
-				$this->imageCount = $this->helper->getImageCount();
-			} else {
-				$this->imageList = $this->helper->getImageList( $this->ts, $do[ 'default' ], $this->order );
-				$this->imageCount = $this->helper->getImageCount( ImageReviewSpecialController::ACTION_UNREVIEWED, count( $this->imageList ) );
-			}
-		} else {
-			$this->imageCount = $this->helper->getImageCount();
-		}
+		$this->imageList = $this->getImageList();
+		$this->imageCount = $this->getImageCounts();
 
 		$this->setVariables();
 
-		$this->imageCount = array_map(function($number) {
-			return $this->wg->Lang->formatNum($number);
-		}, $this->imageCount);
-
 		/* SUS-541 / Mix <mix@wikia.com> / scope: the following if block */
-		$severity = count( $this->imageList ) < ImageReviewHelper::LIMIT_IMAGES ? 'error' : 'success';
+		$severity = count( $this->imageList ) < ImageListGetter::LIMIT_IMAGES
+			? 'error'
+			: 'success';
 		$this->logImageListCompleteness( $severity );
+
+		return true;
 	}
 
-	public function stats() {
-		if ( !$this->wg->User->isAllowed( 'imagereviewstats' )) {
-			$this->specialPage->displayRestrictionError( 'imagereviewstats' );
-			return false;
-		}
+	protected function getImageList() {
+		$imageListGetter = new ImageListGetter(
+			$this->ts,
+			$this::ACTION_TO_STATES[$this->action],
+			$this->order
+		);
+		return $imageListGetter->getImageList();
+	}
 
+	private function getImageCounts() {
+		// Format the number locally (add ',' or '.')
+		return array_map(
+			function( $number ) { return $this->wg->Lang->formatNum( $number ); },
+			$this->imageCountGetter->getImageCounts( $this->wg->User->getId() )
+		);
+	}
+
+	public function stats() {
 		$startDay = $this->request->getVal( 'startDay', date( 'd' ) );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "startDay" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)